pkg/apis/alicloud/validation/shoot.go

```diff
@@ -51,6 +60,22 @@ func ValidateWorkers(workers []core.Worker, zones []apisalicloud.Zone, fldPath *
 			allErrs = append(allErrs, field.Required(fldPath.Index(i).Child("volume"), "must not be nil"))
 		} else {
 			allErrs = append(allErrs, validateVolume(worker.Volume, fldPath.Index(i).Child("volume"))...)
+			if worker.Volume.Encrypted != nil {
+				allErrs = append(allErrs, field.NotSupported(fldPath.Index(i).Child("volume").Child("encrypted"), *worker.Volume.Encrypted, nil))
+			}
+		}
+
+		if length := len(worker.DataVolumes); length > maxDataDiskCount {
+			allErrs = append(allErrs, field.TooMany(fldPath.Index(i).Child("dataVolumes"), length, maxDataDiskCount))
+		}
+		for j, volume := range worker.DataVolumes {
+			dataVolPath := fldPath.Index(i).Child("dataVolumes").Index(j)
+			if volume.Name != nil && !dataDiskNameRegexp.MatchString(*volume.Name) {
+				allErrs = append(allErrs, field.Invalid(dataVolPath.Child("name"), *volume.Name, utilvalidation.RegexError(fmt.Sprintf("disk name given: %s does not match the expected pattern", *volume.Name), dataDiskNameFmt)))
+			} else if volume.Name != nil && len(*volume.Name) > 64 {
+				allErrs = append(allErrs, field.TooLong(dataVolPath.Child("name"), *volume.Name, 64))
+			}
```

Comment:
volume.Name can't be empty for data volumes

Reply:
What do you mean, Jerry? See also https://github.com/gardener/gardener-extension-provider-azure/pull/107#discussion_r438085401

Reply:
> What do you mean, Jerry? See also [gardener/gardener-extension-provider-azure#107 (comment)](https://github.com/gardener/gardener-extension-provider-azure/pull/107#discussion_r438085401)

I just added a comment. If `volume.Name` keeps as a pointer, we need to make sure it is not a `null` value.

Reply:
As said, this is validated by the Gardener already. It cannot reach the system without being set.